### DISCUSSION TITLE
v2.2.0

### DISCUSSION THREAD

**Discussion by LuqueDaniel** [renpy/vscode-language-renpy#327] [Releases]
✨Yeah! ✨

After a long road, we have been able to launch this new version that lays the foundation to develop new features and improve the existing ones.

All thanks to every contributor and user who has helped us to test, validate and improve. Especially to @duckdoom4 whose tokenizer is the centerpiece of this update.

Thank you all!

## What's Changed
* Improved block capture and keywords by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/100
* Enable forced, consistent formatting by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/103
* New feature: styled strings by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/101
* Small fixes in python block and new indenting rules by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/104
* Applied formatting to most files by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/109
* Small fixes by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/117
* Eslint fixes by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/118
* Update vscode engine version and fixed two errors by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/119
* edit: Ignore use of requires, but still show a warning by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/120
* Merge master into new-syntax branch by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/121
* New feature: Document tokenizer by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/112
* Move to its own folder and fix python block end by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/127
* add: Automatic custom color rule injection by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/128
* reg: version 2.0.10 changes by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/129
* fix: Ignore comments in line continuation by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/130
* Pr by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/132
* New feature: Webpack bundler by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/131
* New feature: menu statement by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/102
* Small changes by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/134
* Port back old Ren'Py statements + pause statement by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/135
* New feature: Say statements by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/138
* Improved patterns (label, call and jump) by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/142
* New feature: ATL support by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/143
* [Add] Audio syntax support by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/146
* Adds basic PR validation by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/160
* Prepare v2.1.0 pre-release by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/162
* Fixed line with trailing spaces causing block to break by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/165
* Change `warnOnInvalidVariableNames` config to boolean by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/173
* Fix scene statement highlighting by @Booplicate in https://github.com/LuqueDaniel/vscode-language-renpy/pull/185
* Merge master into develop by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/192
* Pre-release v2.1.2 by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/193
* Allow webpack to execute when installed locally by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/232
* Update python syntax patterns by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/236
* Update the tokenizer patterns by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/235
* Fix/float syntax highlight error by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/237
* Syntax updates by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/239
* Syntax updates by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/240
* Basic screen lang by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/250
* Prepare 2.1.3-pre-release by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/251
* Adds GitHub Actions to Dependabot updates by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/258
* deps: bump eslint-plugin-import from 2.26.0 to 2.27.5 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/262
* deps: bump @typescript-eslint/parser from 5.50.0 to 5.52.0 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/261
* deps: bump @typescript-eslint/eslint-plugin from 5.50.0 to 5.52.0 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/259
* deps: bump @types/node from 18.11.19 to 18.13.0 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/260
* Draft: First check if a key already exists by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/264
* Prepare 2.1.4-pre-release by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/265
* Invert conditional by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/266
* Bump version by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/267
* Update `develop` by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/269
* Properly handle escaped slashes at the end of a string by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/277
* Bump version by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/278
* Fix over extended python line by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/279
* Update custom containers by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/280
* Fix incorrect check by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/281
* Chore: Allow 000*.rpy files in renpy/common by @a2937 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/300
* RunRenpyViaplugin by @seanj29 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/291
* Fix command titles to match convention by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/303
* Ignore dist folder by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/301
* Chore(Enhancement) - Add a command that cleans up compiled files. by @a2937 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/304
* Adds PR Check to master by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/308
* Fixed say statements messing with strings by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/316
* Tokenizer updates by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/317
* Merge master into develop by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/318
* Merge master into develop by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/320
* Merge develop into master (Public Release) by @duckdoom4 in https://github.com/LuqueDaniel/vscode-language-renpy/pull/319
* deps: bump webpack from 5.74.0 to 5.76.0 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/321
* Release 2.2.0 by @LuqueDaniel in https://github.com/LuqueDaniel/vscode-language-renpy/pull/325
* deps: bump eslint from 8.34.0 to 8.42.0 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/322
* deps: bump @typescript-eslint/parser from 5.53.0 to 5.59.8 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/323
* deps: bump @typescript-eslint/eslint-plugin from 5.53.0 to 5.59.8 by @dependabot in https://github.com/LuqueDaniel/vscode-language-renpy/pull/324

## New Contributors
* @Booplicate made their first contribution in https://github.com/LuqueDaniel/vscode-language-renpy/pull/185
* @seanj29 made their first contribution in https://github.com/LuqueDaniel/vscode-language-renpy/pull/291

**Full Changelog**: https://github.com/LuqueDaniel/vscode-language-renpy/compare/v2.0.16...v2.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/LuqueDaniel/vscode-language-renpy/releases/tag/v2.2.0'>v2.2.0</a>.</em>

**Comment by GabingIn3D** [comment]
Hey, ever since this update it seems like parameter colours are displaying incorrectly, my Ren'py scripts are now full of grey blocks of text that 'Developer > Inspect Editor Tokens and Scopes' can only identify as 'other'. My language in VS Code is set to 'Ren'py' and I've also uninstalled and reinstalled VS Code, both standalone and system, and tried the pre-release version of the Ren'py extension, while also disabling/uninstalling all custom colour themes to no avail. This all began when I clicked a 'Reload Required' button on the Ren'py extension today. I'm not sure if this was intentional but I'm having great difficulty working on code now because this seems to be a big hit on its functionality. I included an example of what I'm describing below. The parameters like 'xzoom, yzoom, xalign... etc.' used to be bright blue for me. I also can't seem to auto-type or write screen actions the same way. (please excuse the custom purple theme in the image below, but the same effect is seen in all themes I've tried, including the VS Code defaults) (edit: added another image using the One Dark Pro theme as used in the GIF on the extension page) 
![image](https://github.com/LuqueDaniel/vscode-language-renpy/assets/122047218/3ad6fafd-8178-494f-9788-d104dcddfe2c)
![image](https://github.com/LuqueDaniel/vscode-language-renpy/assets/122047218/b79b51a9-120d-4abd-86f6-ba78467a6789)

**Reply by LuqueDaniel** [reply]
This is due to our latest update, we have implemented a new tokenizer for syntax highlighting. We expect to fix these issues soon. Thank you very much for the report and sorry for the inconvenience.

**Reply by duckdoom4** [reply]
Please try the new version (2.2.1). ATL was accidentally disabled in 2.2.0, let me know if any other issues occur with that version and I'll try to fix them ASAP :)

**Comment by EggnogBaileys** [comment]
Hello! I hope this message finds you well.
I've been using the VS Code Ren'Py extension for quite some time now, but upon opening it today there were quite a few changes that really caught me off-guard. 

For one, parameters on both screens and image definitions don't show up as different colors anymore, neither do the variables (ints or floats right after them). Colors seem to have changed (jump and call labels, anything within semi-colons such as when indenting a sentence in a dialogue string), even when disabling and re-enabling any color theme extensions I had installed. 

Even stranger, entire walls of code now are all the color of strings between quotes, specifically if I were to write a sentence with {i} and not end it with {/i}. This worked fine beforehand, and therefore is structured this way in many of my script files on many lines. 

Also, VS Code's popups which had helped me before (such as when typing a screen action, the popup would tell me what parameters to enter between commas within the parentheses) now do not appear. Also strangely, although I have no idea if this has to do with the update, my VS Code delays when trying to save with CTRL+S rather consistently.

I'm sure this update has a lot added to it but I'm really worried because now my files seem rather odd. Is it still possible to download the prior version? 

Sorry if this comment brought any worry or concern, I want to applaud you for the hard work but with the issues I am experiencing I'm rather worried.

**Reply by LuqueDaniel** [reply]
Sorry for the inconvenience. It would be helpful if you can report all these issues and bugs through the [GitHub issues](https://github.com/LuqueDaniel/vscode-language-renpy/issues), providing code samples and screenshots.

You can install the previous versions, for example, 2.0.16, or even try the pre-release 2.3.1 from the extension page.

![imagen](https://github.com/LuqueDaniel/vscode-language-renpy/assets/1286535/4847515e-c1c1-47f8-ba66-5d48460ac247)

Thank you very much for your feedback.

**Reply by duckdoom4** [reply]
Please try out version 2.2.1, ATL was accidentally disabled in 2.2.0. it should be fixed now and you should see the experience closer to what you were used to. 

Please let us know if you still experience issues with that version and I'll attempt to fix them ASAP :)
